internal/pkg/cli/app_delete.go

```diff
@@ -95,112 +85,153 @@ func (opts deleteAppOpts) Ask() error {
 }
 
 // Execute deletes the application's CloudFormation stack, ECR repository, SSM parameter, and local file.
-func (opts deleteAppOpts) Execute() error {
-	if err := opts.sourceProjectEnvironments(); err != nil {
+func (o *deleteAppOpts) Execute() error {
+	if err := o.initProjectService(o); err != nil {
 		return err
 	}
 
-	if err := opts.deleteStacks(); err != nil {
+	if err := o.sourceProjectEnvironments(); err != nil {
 		return err
 	}
-	if err := opts.emptyECRRepos(); err != nil {
+
+	if err := o.deleteStacks(); err != nil {
+		return err
+	}
+	if err := o.emptyECRRepos(); err != nil {
 		return err
 	}
-	if err := opts.removeAppProjectResources(); err != nil {
+	if err := o.removeAppProjectResources(); err != nil {
 		return err
 	}
-	if err := opts.deleteSSMParam(); err != nil {
+	if err := o.deleteSSMParam(); err != nil {
 		return err
 	}
-	if err := opts.deleteWorkspaceFile(); err != nil {
+	if err := o.deleteWorkspaceFile(); err != nil {
 		return err
 	}
 
-	log.Successf("removed app %s from project %s\n", opts.AppName, opts.projectName)
+	log.Successf("removed app %s from project %s\n", o.AppName, o.projectName)
 	return nil
 }
 
-func (opts *deleteAppOpts) validateAppName() error {
-	localApps, err := opts.workspaceService.Apps()
+func (o *deleteAppOpts) askAppName() error {
+	if o.AppName != "" {
+		return nil
+	}
 
+	// get executed if --name flag is not set
+	names, err := o.retrieveLocalAppName()
+	if err != nil {
+		return err
+	}
+	fmt.Println(names)
```

Comment:
Do we need to print the app names here right before prompting?

Reply:
That should not exist here. Probably a mistake.